internal/testutils/provider.go

```diff
@@ -225,6 +228,22 @@ type MockProviderInfoer struct {
 	GroupsErr bool
 }
 
+// CheckTokenScopes checks if the token has the required scopes.
+func (p *MockProviderInfoer) CheckTokenScopes(token *oauth2.Token) {
+	scopesStr, ok := token.Extra("scope").(string)
+	if !ok {
+		panic(fmt.Sprintf("Failed to cast token scopes to string: %v", token.Extra("scope")))
```

Comment:
also, we could avoid the panic by returning the error here :)

Reply:
I think I would prefer to pass `*testing.T` to `MockProviderInfoer` so that we can use `t.Fatal` in its methods.

Reply:
Sounds good to me!